cilium/network_policy.cc

```diff
@@ -1274,10 +1284,22 @@ NetworkPolicyMap::onConfigUpdate(const std::vector<Envoy::Config::DecodedResourc
         ENVOY_LOG(trace, "Resuming NPDS subscription");
         shared_this->resume();
       } else {
-        ENVOY_LOG_MISC(debug, "NetworkPolicyMap expired on watcher completion!");
+        ENVOY_LOG(debug, "NetworkPolicyMap expired on watcher completion!");
       }
     });
 
+    // Reopen IPcache for every new stream. Cilium agent re-creates IP cache on restart,
+    // and that is also when the old stream terminates and a new one is created.
+    // New security identities (e.g., for FQDN policies) only get inserted to the new IP cache,
+    // so open it before the workers get a chance to enforce policy on the new IDs.
+    if (isNewStream()) {
+      IPCacheSharedPtr ipcache = IPCache::GetIPCache(context_);
+      if (ipcache != nullptr) {
+        ENVOY_LOG(trace, "Reopening ipcache on new stream");
+        ipcache->Open();
```

Comment:
should we do the same check and reset if there is any issue with opening file, also in that case, might need to start subscription for hostmap as well.

Reply:
I think hostmap gets little testing coverage nowadays, would not want to change the operating mode now, on the contrary, I'd like to retire the hostmap ASAP.

But we need to handle the failure case here somehow. I'll try making the lookup reopen when needed.

Reply:
Updated the 1st commit to keep the path and retry open at the beginning of lookup, if it failed. This way we'll recover if there is some temporary failure on opening the map.